### PR TITLE
Rewrite PHPDoc for CommonDBTM properties

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

I tried to rewite PHPDoc to be clean and complete.
I had a lot of remarks on possible improvements, so I put them on TODOs to discuss about it.

### CODE REVIEW

File: inc/commondbtm.class.php [hunk 1]
```diff
@@ -36,64 +36,191 @@
...
+    * @see CommonGLPI::$get_item_to_display_tab
+    */
+   public $get_item_to_display_tab = true;
 
-   /// Search options of the item : to avoid multiple load
-   protected $searchopt                        = false;
+   /**
+    * List of linked item types from plugins on which entities informations should be forwarded on update.
+    *
+    * @TODO Remove this variable as it is not used ?
```
Comment (trasher) [inline inc/commondbtm.class.php]: Use inderectly from CommonDBTM::addForwardEntity() call in Plugin class.

File: inc/commondbtm.class.php [hunk 2]
```diff
@@ -36,64 +36,191 @@
...
+    * @var mixed[]
+    */
+   public $fields = [];
+
+   /**
+    * Flag to determine whether or not changes must be logged into history.
+    *
+    * @var boolean
+    *
+    * @TODO This variable should not be updated and could be converted as a constant (protected if min PHP >= 7.1).
```
Comment (trasher) [inline inc/commondbtm.class.php]: As I've said, I do not agree to change that to constants. Those todo can be removed right now

File: inc/commondbtm.class.php [hunk 3]
```diff
@@ -36,64 +36,191 @@
...
+   protected $fkfield = "";
 
-   /// Set true to desactivate auto compute table name
-   static protected $notable                   = false;
+   /**
+    * Search option of item. Initialized on first call to `self::getOptions()` and used as cache.
+    *
+    * @var array
+    *
+    * @TODO Should be static if used as singleton.
```
Comment (trasher) [inline inc/commondbtm.class.php]: If this is used as cache this should be dropped in favor of real cache.

File: inc/commondbtm.class.php (hunk 2 above)
Comment (cedric-anne) [reply inc/commondbtm.class.php]: OK.

File: inc/commondbtm.class.php (hunk 1 above)
Comment (cedric-anne) [reply inc/commondbtm.class.php]: Yep. I missed it.

File: inc/commondbtm.class.php (hunk 3 above)
Comment (cedric-anne) [reply inc/commondbtm.class.php]: TODO Updated.

File: inc/commondbtm.class.php (hunk 2 above)
Comment (cedric-anne) [reply inc/commondbtm.class.php]: I changed for another proposal. Can be refused too :).

**Pull request merged**